packages/starknet_provider/test/integration/read_provider_test.dart

```diff
@@ -1116,34 +1116,160 @@ void main() {
+      });
+    });
+
+    group('getStorageProof', () {
+      test(
+          'returns storage proof for valid contract addresses and storage keys',
+          () async {
+        if (!await isRpcVersionSufficient(provider, '0.8')) {
+          return;
+        }
+        final contractAddresses = [
+          Felt.fromHexString('0x1234567890123456789012345678901234567890'),
+        ];
+        final contractsStorageKeys = [
+          ContractStorageKeys(
+            contractAddress: Felt.fromHexString(
+                '0x1234567890123456789012345678901234567890'),
+            storageKeys: [
+              Felt.fromHexString('0x0987654321098765432109876543210987654321'),
+            ],
+          ),
+        ];
+
+        final request = GetStorageProofRequest(
+          blockId: BlockId.latest,
+          contractAddresses: contractAddresses,
+          contractsStorageKeys: contractsStorageKeys,
+        );
+
+        final response = await provider.getStorageProof(request);
+
+        response.when(
+          error: (error) => fail('Should not fail: ${error.message}'),
+          result: (result) {
+            expect(result.classesProof, isA<Map<String, dynamic>>());
+            expect(result.contractsProof, isA<ContractsProof>());
+            expect(result.contractsStorageProofs,
+                isA<List<Map<String, dynamic>>>());
+            expect(result.globalRoots, isA<GlobalRoots>());
+            expect(result.globalRoots.contractsTreeRoot, isNotNull);
+            expect(result.globalRoots.classesTreeRoot, isNotNull);
+            expect(result.globalRoots.blockHash, isNotNull);
+          },
+        );
+      });
+
+      test('returns error with empty request', () async {
+        if (!await isRpcVersionSufficient(provider, '0.8')) {
+          return;
+        }
+        final request = GetStorageProofRequest(
+          blockId: BlockId.latest,
+        );
+
+        final response = await provider.getStorageProof(request);
+
+        response.when(
+          error: (error) {
+            expect(error.code, JsonRpcApiErrorCode.INVALID_QUERY);
+          },
+          result: (result) => fail('Should fail'),
+        );
+      });
+
+      test('returns error with invalid block id', () async {
+        if (!await isRpcVersionSufficient(provider, '0.8')) {
+          return;
+        }
+        final contractAddresses = [
+          Felt.fromHexString('0x1234567890123456789012345678901234567890'),
+        ];
+
+        final request = GetStorageProofRequest(
+          blockId: invalidBlockIdFromBlockHash,
+          contractAddresses: contractAddresses,
+        );
+
+        final response = await provider.getStorageProof(request);
+
+        response.when(
+          error: (error) {
+            expect(error.code, JsonRpcApiErrorCode.BLOCK_NOT_FOUND);
+            expect(error.message, 'Block not found');
+          },
+          result: (result) => fail('Should fail'),
+        );
+      });
+    });
+
```

Comment:
Hum it looks like you have duplicated these tests: maybe when trying to resolve conflict.